R/zchunk_L2322.Fert.R

```diff
@@ -76,272 +76,447 @@ module_energy_L2322.Fert_DISABLED <- function(command, ...) {
+      select(LEVEL2_DATA_NAMES[["StubTechProd"]]) ->
+      L2322.StubTechProd_Fert
+
+    # L2322.StubTechCoef_Fert: calibrated base-year coefficients of fertilizer production technologies
+    L1322.IO_R_Fert_F_Yh %>%
+      filter(year %in% BASE_YEARS) %>%
+      rename(coefficient = value) %>%
+      mutate(coefficient = round(coefficient, energy.DIGITS_COEFFICIENT)) %>%
+      filter(coefficient != 0) %>% # Where 0, drop from this table (to revert to assumed defaults)
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID") %>%
+      left_join_error_no_match(select(calibrated_techs, sector, fuel, supplysector, subsector, technology, minicam.energy.input), by = c("sector", "fuel")) %>%
+      mutate(stub.technology = technology, market.name = region) %>%
+      select(LEVEL2_DATA_NAMES[["StubTechCoef"]]) ->
+      L2322.StubTechCoef_Fert
+
+    # L2322.StubTechFixOut_Fert_imp: fixed output of import technology (fixed imports)
+    # Imports are negative net exports
+    L142.ag_Fert_NetExp_MtN_R_Y %>%
+      filter(year %in% BASE_YEARS) %>%
+      rename(fixedOutput = value) %>%
+      left_join_error_no_match(GCAM_region_names, by = 'GCAM_region_ID') %>%
+      mutate(supplysector = A322.globaltech_renew[["supplysector"]], subsector = A322.globaltech_renew[["subsector"]], stub.technology = A322.globaltech_renew[["technology"]]) %>%
+      mutate(fixedOutput = pmax(0, -1 * fixedOutput)) %>%
+      mutate(fixedOutput = round(fixedOutput, energy.DIGITS_CALOUTPUT)) %>%
+      mutate(share.weight.year = year, subs.share.weight = 0, tech.share.weight = 0) %>%
+      bind_rows(repeat_add_columns(select(filter(., year == max(BASE_YEARS)), -year), tibble(year = FUTURE_YEARS))) %>%
+      select(LEVEL2_DATA_NAMES[["StubTechFixOut"]]) -> #Repeat final year to all future years and rbind
+      L2322.StubTechFixOut_Fert_imp
+
+    # L2322.StubTechFixOut_Fert_exp: fixed output of import technology (fixed imports)
+    # Exports are positive net exports
+    L142.ag_Fert_NetExp_MtN_R_Y %>%
+      filter(year %in% BASE_YEARS) %>%
+      rename(fixedOutput = value) %>%
+      mutate(fixedOutput = round(fixedOutput, energy.DIGITS_CALOUTPUT)) %>%
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID" ) %>%
+      mutate(supplysector = filter(A322.globaltech_shrwt, grepl( "Exports", supplysector))[["supplysector"]]) %>%
+      mutate(subsector = filter(A322.globaltech_shrwt, grepl( "Exports", supplysector))[["subsector"]]) %>%
+      mutate(stub.technology = filter(A322.globaltech_shrwt, grepl( "Exports", supplysector))[["technology"]]) %>%
+      mutate(fixedOutput = pmax(0, fixedOutput), share.weight.year = year, subs.share.weight = 0, tech.share.weight = 0) ->
+      L2322.StubTechFixOut_Fert_exp_base
+
+    L2322.StubTechFixOut_Fert_exp_base %>%
```

Comment:
Can't we combine this pipeline with the previous?

Reply:
`L2322.StubTechFixOut_Fert_exp_base` will be used later to generate another tibble.